File: CodeWars/7kyu/Highest_and_Lowest.cpp
/*
In this little assignment you are given a string of space separated numbers, and have to return the highest and lowest number.

Examples
Input: "1 2 3 4 5"   =>  Output: "5 1"
Input: "1 2 -3 4 5"  =>  Output: "5 -3"
Input: "1 9 3 4 -5"  =>  Output: "9 -5"
Notes
All numbers are valid Int32, no need to validate them.
There will always be at least one number in the input string.
Output string must be two numbers separated by a single space, and highest number is first.
FUNDAMENTALSSTRINGS
*/
// Soliution
#include <string>
#include <sstream>

std::string highAndLow(const std::string& numbers){
  std::istringstream iss(numbers);
  int min_number = 10e7;
  int max_number = -10e7;
  int number;
  while (iss >> number) {
    min_number = std::min(min_number, number);
    max_number = std::max(max_number, number);
  }
  return std::to_string(max_number) + " " + std::to_string(min_number);
}

File: CodeWars/7kyu/Highest_and_Lowest.go
/*
In this little assignment you are given a string of space separated numbers, and have to return the highest and lowest number.

Examples
HighAndLow("1 2 3 4 5") // return "5 1"
HighAndLow("1 2 -3 4 5") // return "5 -3"
HighAndLow("1 9 3 4 -5") // return "9 -5"
Notes
All numbers are valid Int32, no need to validate them.
There will always be at least one number in the input string.
Output string must be two numbers separated by a single space, and highest number is first.
FundamentalsStrings
*/
// Solution
package kata

import "strconv"

func HighAndLow(in string) string {
  var mn, mx, acc int = 1e5, -1e5, 0
  var isNeg bool = false;
  for _, ch := range in {
    if ch == ' ' {
      if isNeg {
        acc *= -1
      }
      mn = min(mn, acc)
      mx = max(mx, acc)
      acc = 0
      isNeg = false
    } else if ch == '-' {
      isNeg = true
    } else {
      acc = acc * 10 + int(ch - '0')
    }
  }
  if isNeg {
    acc *= -1
  }
  mn = min(mn, acc)
  mx = max(mx, acc)
  return strconv.Itoa(mx) + " " + strconv.Itoa(mn)
}